server/src/main/java/org/elasticsearch/gateway/LocalAllocateDangledIndices.java

```diff
@@ -149,7 +151,7 @@ public ClusterState execute(ClusterState currentState) {
                         }
                         metaData.put(upgradedIndexMetaData, false);
                         blocks.addBlocks(upgradedIndexMetaData);
-                        if (upgradedIndexMetaData.getState() == IndexMetaData.State.OPEN) {
+                        if (upgradedIndexMetaData.getState() == IndexMetaData.State.OPEN || isIndexVerifiedBeforeClosed(indexMetaData)) {
```

Comment:
I think we can remove this check here. We have the same check inside method `addAsFromDangling`.

Reply:
In order to keep things aligned with other usages, I prefer to keep this as is for now.